sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -437,61 +437,76 @@ object DataSourceStrategy {
     }
   }
 
+  /**
+   * Find the column name of an expression that can be pushed down.
+   */
+  private[sql] def pushDownColName(e: Expression): Option[String] = {
```

Comment:
Shall we make it an object with `unapply`, and use it as a pattern match below?

Reply:
With 
```scala
object PushDownColName {
  def unapply(e: Expression): Option[String] = {
    def helper(e: Expression) = e match {
      case a: Attribute => Some(a.name)
      case _ => None
    }
    helper(e)
  }
}
```
The following code can be written
```scala
    case expressions.EqualTo(PushDownColName(name), Literal(v, t)) =>
      Some(sources.EqualTo(name, convertToScala(v, t)))
```
instead of 
```scala
    case expressions.EqualTo(e: Expression, Literal(v, t)) =>
      pushDownColName(e).map(sources.EqualTo(_, convertToScala(v, t)))
```

I don't have strong preference about it.  What do others feel?

Reply:
+1 for @HyukjinKwon 's suggestion and the new @dbtsai 's code.

Reply:
I was thinking the pattern match seems better with less diff and more consistent. There's also one similar example such as https://github.com/apache/spark/blob/0032d85153e34b9ac69598b7dff530094ed0f640/sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala#L194-L203

It might be best to rename `PushDownColName` something like `PushableColumnName` but no strong preference.

Reply:
Oh, thanks, @dongjoon-hyun. I didn't see your comment before my comment.

Reply:
Ur, the last commit is quite different from the above sample code. I'm not sure that last commit is better or not.

Reply:
I re-named it to make it cleaner.